### ISSUE TITLE
Calling flush on LineProtoSender sometimes causes io.questdb.network.NetworkError: [10053] send error

### ISSUE THREAD

**Issue by sarunas-stasaitis** [questdb/questdb#1076]
**Describe the bug**

```
io.questdb.network.NetworkError: [10053] send error
	at io.questdb.std.ThreadLocal.initialValue(ThreadLocal.java:36)
	at java.base/java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:195)
	at java.base/java.lang.ThreadLocal.get(ThreadLocal.java:172)
	at io.questdb.network.NetworkError.instance(NetworkError.java:47)
	at io.questdb.cutlass.line.tcp.LineTCPProtoSender.sendToSocket(LineTCPProtoSender.java:55)
	at io.questdb.cutlass.line.LineProtoSender.send(LineProtoSender.java:275)
	at io.questdb.cutlass.line.LineProtoSender.flush(LineProtoSender.java:156)
	at lt.energyadvice.qdbtest.MainQuestDB.main(MainQuestDB.java:73)
	at org.codehaus.mojo.exec.ExecJavaMojo$1.run(ExecJavaMojo.java:254)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**To Reproduce**
```
import io.questdb.cutlass.line.LineProtoSender;
import io.questdb.cutlass.line.tcp.LineTCPProtoSender;
import io.questdb.network.Net;
import org.apache.commons.csv.CSVFormat;
import org.apache.commons.csv.CSVRecord;

import java.io.File;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.nio.charset.StandardCharsets;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.concurrent.atomic.AtomicLong;

import static lt.energyadvice.qdbtest.Util.dtf;
import static lt.energyadvice.qdbtest.Util.sleep;

public class MainQuestDB {

    private static final AtomicLong count = new AtomicLong();

    public static void main(String[] args) throws Exception {

        File file = new File("path_to_csv");

        String hostIPv4 = "127.0.0.1";
        int port = 9009;
        int bufferCapacity = 256*1024*1024;

        System.out.println("Connected");

        String dbName = "ksImport" + System.currentTimeMillis();

        Thread t = new Thread(() -> {
            long prev = 0;
            while (true) {
                long l = count.get();
                System.out.println(l - prev);
                prev = l;
                sleep(1000);
            }
        });
        t.setDaemon(true);
        t.start();

        long start = System.currentTimeMillis();

        try (
            LineProtoSender sender = new LineTCPProtoSender(Net.parseIPv4(hostIPv4), port, bufferCapacity);
            InputStreamReader ir = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8)
        ) {
            int c = 0;
            for (CSVRecord record : CSVFormat.EXCEL.withFirstRecordAsHeader().parse(ir)) {
                long datasetId = Long.parseLong(record.get("DATASET_ID"));
                Instant time = LocalDateTime.parse(record.get("IDENTIFIER"), dtf).toInstant(ZoneOffset.UTC);
                double v;
                try {
                    v = Double.parseDouble(record.get("VALUE"));
                } catch (Exception ex) {
                    continue;
                }

                sender.metric(dbName + "_" + datasetId)
                    .field("VALUE", v)
                    .$(time.toEpochMilli() * 1000000);

                count.incrementAndGet();
                if (++c % 100000 == 0) {
                    sender.flush();
                }
            }
            sender.flush();
        } catch (Throwable ex) {
            ex.printStackTrace();
        } finally {
            System.out.println(((System.currentTimeMillis() - start) / 1000) + " s");
            System.exit(0);
        }

    }

}

```

**Expected behavior**
Code completes with no errors

**Environment (please complete the following information):**
 - OS: Client: Java 11 on Windows, Server: questdb docker image running on windows docker
 - Version: 6.0.2, 6.0.3

**Comment by ideoma** [comment]
Thanks for reporting the bug. Would be useful if you look at server log at that time to see if server aborts the connection or any other type of error

```
docker <container_name/id> logs
```

**Comment by sarunas-stasaitis** [comment]
Got a slightly different error this time: io.questdb.network.NetworkError: [10054] send error

The issue occured arround 2021-06-01 15:02 UTC

```
{"log":"2021-06-01T15:02:18.394125Z E i.q.c.l.t.LineTcpConnectionContext [37] could not process line data\r\n","stream":"stdout","time":"2021-06-01T15:02:18.3951678Z"}
{"log":"java.lang.NullPointerException\r\n","stream":"stdout","time":"2021-06-01T15:02:18.3951926Z"}
```

seems related, could be wrong, full log attached

[questdb.log](https://github.com/questdb/questdb/files/6582257/questdb.log)

**Comment by ideoma** [comment]
Thanks for sending the log file, really helpful.

There was a bug which caused NullPointerException with the symptoms you described, it was fixed in 6.0.3.
Your logs indicate that you ran 6.0.2 when the error happened (all messages in the log file are from 6.0.2)

Could you confirm it's still a problem with 6.0.3 and perhaps give an updated log?

**Comment by sarunas-stasaitis** [comment]
Issue not present in 6.0.3
